### PR TITLE
TicTacToe from Miri

### PR DESCRIPTION
**Pull request by miribar**

### CODE REVIEW

File: Grades.java [hunk 1]
```diff
@@ -1,18 +0,0 @@
...
- * 1. The highest grade and its index (something like: the student #10 got the highest grade and it's 98).
- * 2. The lowest grade and its index (something like: the student #3 got the lowest grade and it's 13).
- * 3. The average of the grades
- */
-public class Grades {
-
-    public static void main(String[] args) {
-        // enter your code here :)
-    }
-}
```
Comment (miribar) [inline Grades.java:18]: Hi @ameedf Please ignore the clutter... had to send this pull request from my work laptop, without my git repo.
Thanks!

File: Player.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package TicTacToe;
+
+class Player {
```
Comment (ameedf) [inline Player.java:3]: You forgot the "public" modifier - as I said in the class: for now, only public classes are permitted

File: Player.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    private final char playerMark;
+    private final String playerName;
+    private boolean isNextPlayer;
+
+    Player(char playerMark, String playerName) {
+        this.playerMark = playerMark;
+        this.playerName = playerName;
+    }
+
+    char getPlayerMark() {
```
Comment (ameedf) [inline Player.java:14]: I guess that you meant "public" also here

File: TicTacToe.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            }
+        }
+    }
+
+    boolean tickGameBoard(byte row, byte column, char playerMark) {
+        if (this.gameBoard[row][column] == ' ') {
+            gameBoard[row][column] = playerMark;
+            return true;
+        }
+        else
```
Comment (ameedf) [inline TicTacToe.java:20]: no need to write "else"

File: TicTacToe.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        {
+            System.out.print("This cell has already been selected, please select another cell: ");
```
Comment (ameedf) [inline TicTacToe.java:22]: The syso action shouldn't be performed here

File: TicTacToe.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            return playerX;
+        }
+        else {
+            playerO.setNextPlayer(true);
+            playerX.setNextPlayer(false);
+            return playerO;
+        }
+    }
+
+    boolean checkIfGameWon() {
```
Comment (ameedf) [inline TicTacToe.java:51]: 👍 for the code breaking

**Review by ameedf — COMMENTED** [review]
Very good job..
Can you think of an implementation where you don't have to store the "playerMark" and "isNextPlayer" in the Player class ? actually you might want to totally remove this class

File: Player.java (hunk 2 above)
Comment (miribar) [reply Player.java:3]: True, we haven't learned package-level entry yet.

File: TicTacToe.java (hunk 4 above)
Comment (miribar) [reply TicTacToe.java:20]: True. I'll revise.

**Comment by miribar** [comment]
Thank you for the review @ameedf 😃

**Pull request closed without merging**